src/components/coin.ts

```diff
@@ -72,3 +96,25 @@ export const adjustCoinBalanceByUserId = async (userId: string, amount: number):
     amount
   );
 };
+
+/*
+  reason is only applicable for admin commands and is optional.
+  adminId is only applicable for admin commands and is mandatory.
+*/
+export const updateUserCoinLedger = async (
```

Comment:
maybe rename to `createCoinLedgerEntry`, i think update might be misleading here

Reply:
function docstring should also explain what the function does on a high level